rust/arrow/src/array.rs

```diff
@@ -234,6 +235,42 @@ pub struct PrimitiveArray<T: ArrowPrimitiveType> {
     raw_values: RawPtrBox<T::Native>,
 }
 
+/// Common operations for primitive types, including numeric types and boolean type.
+pub trait PrimitiveArrayOps<T: ArrowPrimitiveType> {
+    fn values(&self) -> Buffer;
+    fn value(&self, i: usize) -> T::Native;
+}
+
+impl<T: ArrowPrimitiveType> PrimitiveArrayOps<T> for PrimitiveArray<T> {
+    default fn values(&self) -> Buffer {
+        unimplemented!()
+    }
+
+    default fn value(&self, _: usize) -> T::Native {
+        unimplemented!()
+    }
+}
+
+impl<T: ArrowNumericType> PrimitiveArrayOps<T> for PrimitiveArray<T> {
+    fn values(&self) -> Buffer {
+        self.values()
+    }
+
+    fn value(&self, i: usize) -> T::Native {
+        self.value(i)
+    }
+}
+
+impl PrimitiveArrayOps<BooleanType> for BooleanArray {
+    fn values(&self) -> Buffer {
+        self.values()
```

Comment:
The implementation of boolean type is same to numeric type, why we need another implementation block?

Reply:
This is because we implemented `PrimitiveArray` separately for `ArrowNumericType` and `BooleanType`. Even though they have the same method signature, we cannot access the method under the `PrimitiveArray` trait.